src/Symfony/Component/Config/Loader/FileLoader.php

```diff
@@ -59,23 +59,43 @@ public function getLocator()
     /**
      * Imports a resource.
      *
-     * @param mixed       $resource       A Resource
-     * @param string|null $type           The resource type or null if unknown
-     * @param bool        $ignoreErrors   Whether to ignore import errors or not
-     * @param string|null $sourceResource The original resource importing the new resource
+     * @param mixed                $resource       A Resource
+     * @param string|null          $type           The resource type or null if unknown
+     * @param bool                 $ignoreErrors   Whether to ignore import errors or not
+     * @param string|null          $sourceResource The original resource importing the new resource
+     * @param string|string[]|null $exclude        A globbed path of files to exclude or an array of globbed paths of files to exclude
      *
      * @return mixed
      *
      * @throws LoaderLoadException
      * @throws FileLoaderImportCircularReferenceException
      * @throws FileLocatorFileNotFoundException
      */
-    public function import($resource, $type = null, $ignoreErrors = false, $sourceResource = null)
+    public function import($resource, $type = null, $ignoreErrors = false, $sourceResource = null /* , $exclude = null */)
     {
+        if (\func_num_args() < 5 && __CLASS__ !== \get_class($this) && __CLASS__ !== (new \ReflectionMethod($this, __FUNCTION__))->getDeclaringClass()->getName() && !$this instanceof \PHPUnit\Framework\MockObject\MockObject && !$this instanceof \Prophecy\Prophecy\ProphecySubjectInterface) {
+            @trigger_error(sprintf('The "%s()" method will have a new "$exclude = null" argument in version 5.0, not defining it is deprecated since Symfony 4.3.', __METHOD__), E_USER_DEPRECATED);
+        }
+
         if (\is_string($resource) && \strlen($resource) !== $i = strcspn($resource, '*?{[')) {
+            $excludePrefix = null;
+            $exclude = (func_num_args() >= 5) ? \func_get_arg(5) : null;
+            $excludePatterns = (array) $exclude;
+            $excludePaths = [];
+            foreach ($excludePatterns as $excludePattern) {
+                foreach ($this->glob($excludePattern, true, $_, false, true) as $path => $info) {
+                    if (null === $excludePrefix) {
+                        $excludePrefix = $_->getPrefix();
+                    }
+
+                    // normalize Windows slashes
+                    $excludePaths[str_replace('\\', '/', $path)] = true;
+                }
+            }
+
             $ret = [];
             $isSubpath = 0 !== $i && false !== strpos(substr($resource, 0, $i), '/');
-            foreach ($this->glob($resource, false, $_, $ignoreErrors || !$isSubpath) as $path => $info) {
+            foreach ($this->glob($resource, false, $_, $ignoreErrors || !$isSubpath, false, (array) $excludePaths) as $path => $info) {
```

Comment:
you don't need to cast to array here, you know for sure that `$excludePaths` is already an array